src/parser/statement.js

```diff
@@ -303,35 +304,38 @@ pp.parseForStatement = function (node) {
     return this.parseFor(node, init);
   }
 
+  let usedNowKeyword = false;
+  if (this.hasPlugin("lightscript") && (!forAwait)) {
+    // idx, elem, key, or val begins a LS-enhanced loop.
+    if (
+      (this.isContextual("idx") || this.isContextual("elem") || this.isContextual("key") || this.isContextual("val"))
+    ) {
+      return this.parseEnhancedForIn(node);
+    }
+
+    // Possibly for now...in
+    usedNowKeyword = this.eat(tt._now);
```

Comment:
Normally `now x` on its own raises `"now can only be used with assignments and updates."`, which you'll see just below the line I linked. We want to allow it in `for-in` and `for-of` loops, though, which conveniently is already denoted by the `noIn` parameter in `parseExpression`.